pandas/_libs/parsers.pyx

```diff
@@ -702,7 +702,7 @@ cdef class TextReader:
                         ic = (len(self.index_col) if self.index_col
                               is not None else 0)
 
-                        if lc != unnamed_count and lc - ic > unnamed_count:
+                        if (lc != unnamed_count and lc - ic > unnamed_count) or ic == 0:
```

Comment:
can you add a similar comment to below (e.g. for future readers what are the inference rules)